mapper-processor/src/main/java/com/datastax/oss/driver/internal/mapper/processor/entity/EntityHelperSchemaValidationMethodGenerator.java

```diff
@@ -0,0 +1,309 @@
+    else if (targetElement.get().equals(TargetElement.TABLE)) {
+      findMissingColumnsInTable(methodBuilder);
+    }
+    // if explicitly provided SchemaHint is UDT, then generate only UDT check
+    else if (targetElement.get().equals(TargetElement.UDT)) {
+      findMissingColumnsInUdt(methodBuilder, false);
+    }
+  }
+
+  // Finds out missingTableCqlNames - columns that are present in Entity Mapping but NOT present in
+  // CQL table
+  private void findMissingColumnsInTable(MethodSpec.Builder methodBuilder) {
+    methodBuilder.beginControlFlow("if (tableMetadata.isPresent())");
+
+    methodBuilder.addComment("validation of missing Clustering Columns");
+    generateMissingClusteringColumnsCheck(methodBuilder);
+
+    methodBuilder.addComment("validation of missing PKs");
+    generateMissingPKsCheck(methodBuilder);
+
+    methodBuilder.addComment("validation of all columns");
+    generateMissingColumnsCheck(methodBuilder);
+
+    methodBuilder.addComment("validation of types");
+    generateColumnsTypeCheck(methodBuilder);
+
+    methodBuilder.endControlFlow();
+  }
+
+  private void generateColumnsTypeCheck(MethodSpec.Builder methodBuilder) {
+    methodBuilder.addStatement(
+        "$1T<$2T, $3T<?>> expectedTypesPerColumn = new $4T<>()",
+        Map.class,
+        CqlIdentifier.class,
+        GenericType.class,
+        LinkedHashMap.class);
+
+    Map<CodeBlock, TypeName> expectedTypesPerColumn =
+        entityDefinition.getAllColumns().stream()
+            .collect(
+                Collectors.toMap(PropertyDefinition::getCqlName, v -> v.getType().asRawTypeName()));
+
+    for (Map.Entry<CodeBlock, TypeName> expected : expectedTypesPerColumn.entrySet()) {
+      methodBuilder.addStatement(
+          "expectedTypesPerColumn.put($1T.fromCql($2L), $3L)",
+          CqlIdentifier.class,
+          expected.getKey(),
+          entityHelperGenerator.addGenericTypeConstant(expected.getValue().box()));
+    }
+
+    methodBuilder.addStatement(
+        "$1T<$2T> missingTableTypes = findMissingTypes(expectedTypesPerColumn, (($3T) tableMetadata.get()).getColumns(), context.getSession().getContext().getCodecRegistry())",
+        List.class,
+        String.class,
+        DefaultTableMetadata.class);
+    methodBuilder.addStatement(
+        "throwMissingTypesIfNotEmpty(missingTableTypes, keyspaceId, tableId, entityClassName)");
+  }
+
+  private void generateMissingColumnsCheck(MethodSpec.Builder methodBuilder) {
+
+    methodBuilder.addStatement(
+        "$1T<$2T> missingTableCqlNames = findMissingColumnsCql(expectedCqlNames, (($3T) tableMetadata.get()).getColumns().keySet())",
+        List.class,
+        CqlIdentifier.class,
+        DefaultTableMetadata.class);
+
+    // Throw if there are any missingTableCqlNames
+    CodeBlock missingCqlColumnExceptionMessage =
+        CodeBlock.of(
+            "String.format(\"The CQL ks.table: %s.%s has missing columns: %s that are defined in the entity class: %s\", "
```

Comment:
```suggestion
            "String.format(\"The CQL table: %s.%s has missing columns: %s that are defined in the entity class: %s\", "
```
?


Also occurs a couple of other times below.

Reply:
why do you want to remove ks from this message? I think it is important information, the same table name could be present in multiple keyspaces, and putting ks in the log will give the user more context.

Reply:
If you just say "table" or "udt" it's enough to indicate the type of entity.

I see the intent now, but it looks a bit unusual in a plain english message. Also if you just say "the table inventory.product", it should be pretty obvious enough that "inventory" is the keyspace.

But that's just a cosmetic detail so I don't want to dwell on it too much, I'll leave the final decision to you.

Reply:
When writing tests and debugging it was useful to me have that `ks.udt` and `table.udt` in logs so I will leave those in the logs if you don't mind